### PR TITLE
First support for testing containers.

### PR DESCRIPTION
**Pull request by phracek**
First support for testing containers.
The new engines are created.
container.py is used for manipulation with containers w/o S2I support. s2i_container.py is used for manipulation with containers with s2i support. Podman Wrapper is used for working with podman / docker binaries

### CODE REVIEW

File: container_ci_suite/engines/container.py [hunk 1]
```diff
@@ -21,69 +21,508 @@
...
+        df_name = Path(ntf)
+        df_content = self.s2i_create_df(
+            tmp_dir=tmp_dir,
+            app_path=app_path,
+            s2i_args=s2i_args,
+            src_image=src_image,
+            dst_image=dst_image,
         )
+        with open(df_name, mode="w") as f:
+            f.writelines(df_content)
```
Comment (SlouchyButton) [inline container_ci_suite/engines/container.py:98]: https://docs.python.org/3/library/io.html#io.IOBase.writelines
'Line separators are not added, so it is usual for each of the lines provided to have a line separator at the end.'

Not adding `\n` to every line in `df_content` will result in the 'Error: FROM requires either one argument, or three: FROM <source> [AS <name>]' - the resulting dockerfile will have one single line only.

File: container_ci_suite/engines/container.py [hunk 2]
```diff
@@ -21,69 +21,508 @@
...
+            user = "0"
+
+        assert int(user)
+        user_id = PodmanCLIWrapper.docker_get_user_id(src_image=src_image, user=user)
+        if not user_id:
+            logger.error(f"id of user {user} not found inside image {src_image}.")
+            logger.error("Terminating s2i build.")
+            return None
+        else:
+            user_id = user
```
Comment (SlouchyButton) [inline container_ci_suite/engines/container.py:132]: The `else` statement is redundant. Also, `user` variable can (in my testing) contain a newline at the end, which broke the dockerfile.

I'd remove `else` statement and use `.strip()` in `docker_get_user_id` function in `podman_wrapper.py`.

_Strip function probably should be in the `docker_inspect` function as well. Can't think of a reason why we would want output with trailing/leading whitespaces._

Comment (phracek) [reply container_ci_suite/engines/container.py]: Good point. 'else' is removed by this commit https://github.com/sclorg/container-ci-suite/pull/93/commits/9a165e38ebd34b3d2826faaebd8710e99ab03b1e

File: container_ci_suite/engines/container.py (hunk 1 above)
Comment (phracek) [reply container_ci_suite/engines/container.py]: Function `s2i_create_df` returns list of `List[str]`. But I will re-write it a bit. 
Commit https://github.com/sclorg/container-ci-suite/pull/93/commits/84e5eb6caa17b7ba2eea078a8292268b46efa952

**Comment by phracek** [comment]
Let's get merge it so the PR is not so big.

**Review by SlouchyButton — APPROVED** [review]

**Pull request merged**